app/src/main/java/ch/epfl/culturequest/ui/home/HomeViewModel.java

```diff
@@ -0,0 +1,19 @@
+package ch.epfl.culturequest.ui.home;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
```

Comment:
can you add some doc? like what the class does and how it's gonna be used?

Reply:
For now it is empty so there is nothing to comment. But to summarize it quickly: A fragment is like like an activity but the difference is that is hosted by an activity or another fragment and cannot live on its own. Why do we use it ? Because like that for example we can display a static menu on a single activity and still navigate through fragments to change only a part of the screen while always keeping the menu displayed.

Then why do we use ViewModel ? A ViewModel is not mandatory here but it is good practice because it keeps all the data in memory when we close a fragment and it also allows to manage the values displayed in the UI outside of the fragment which is better. Why it is better ? Because Fragment and activities should normally only contain the logic and not the UI data managing stuff.

Finally what is Binding ? Binding is a support library that allows you to bind UI components in your layouts to data sources in your app using a declarative format rather than programmatically, i.e instead of using findViewById(). Why is it better ? Because it allows to observe data changes easily and to update Views on the fly while also detecting Null Pointer exceptions during build time instead of runtime which is better for testing.

If you want more details about all that, you can find everything in the android doc here : https://developer.android.com/guide

Reply:
Thanks for the explanation!! much clearer now